### PR TITLE
Use Serialized OIDs to identify OpenSSL

### PR DESCRIPTION
**Pull request by Ana06**
Use Serialized OIDs to identify RSA, DSA and ECDSA from OpenSSL.

Reference:
- https://github.com/openssl/openssl/blob/fdc5043d58900663b493147298e64f11353b35fe/crypto/objects/obj_dat.h

Identifying the bytes `2A 86 48 86 F7 0D 01 01 01` was key to identify the encryption in a Rust sample I have recently analysed. Next time capa will speed me up. 😄 I can not share the sample, so adding these rules to the nursery.

### CODE REVIEW

File: nursery/encrypt-data-using-openssl-dsa.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+rule:
+  meta:
+    name: encrypt data using openssl DSA
```
Comment (williballenthin) [inline nursery/encrypt-data-using-openssl-dsa.yml:3]:
```suggestion
    name: encrypt data using OpenSSL DSA
```

File: nursery/encrypt-data-using-openssl-ecdsa.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+rule:
+  meta:
+    name: encrypt data using openssl ECDSA
```
Comment (williballenthin) [inline nursery/encrypt-data-using-openssl-ecdsa.yml:3]:
```suggestion
    name: encrypt data using OpenSSL ECDSA
```

File: nursery/encrypt-data-using-openssl-rsa.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+rule:
+  meta:
+    name: encrypt data using openssl RSA
```
Comment (williballenthin) [inline nursery/encrypt-data-using-openssl-rsa.yml:3]:
```suggestion
    name: encrypt data using OpenSSL RSA
```

File: nursery/encrypt-data-using-openssl-dsa.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    namespace: data-manipulation/encryption/dsa
+    authors:
+      - "Ana06"
+    scope: function
+    references:
+      - https://github.com/openssl/openssl/blob/fdc5043d58900663b493147298e64f11353b35fe/crypto/objects/obj_dat.h
+  features:
+    - and:
+      - os: linux
```
Comment (williballenthin) [inline nursery/encrypt-data-using-openssl-dsa.yml:12]: i think this code can be statically linked into programs on Windows/MacOS/etc. too, right? so can we remove this check?

**Review by williballenthin — APPROVED** [review]
see inline comments, otherwise, great!

**Review by mr-tz — COMMENTED** [review]
looks good, thank you!

File: nursery/encrypt-data-using-openssl-rsa.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        - bytes: 2A 86 48 86 F7 0D 01 01 03 = RSA-MD4
+        - bytes: 2A 86 48 86 F7 0D 01 01 0B = RSA-SHA256
+        - bytes: 2A 86 48 86 F7 0D 01 01 0C = RSA-SHA384
+        - bytes: 2A 86 48 86 F7 0D 01 01 0D = RSA-SHA512
+        - bytes: 2A 86 48 86 F7 0D 01 01 0E = RSA-SHA224
+        - bytes: 2A 81 1C CF 55 01 83 78 = RSA-SM3
+        - bytes: 2A 86 48 86 F7 0D 01 01 0F = RSA-SHA512/224
+        - bytes: 2A 86 48 86 F7 0D 01 01 10 = RSA-SHA512/256
+        - bytes: 55 08 03 64 = RSA-MDC2
+        - bytes: 2B 24 03 03 01 0 = RSA-RIPEMD160
```
Comment (mr-tz) [inline nursery/encrypt-data-using-openssl-rsa.yml:30]: invalid byte sequence (see linter)

File: nursery/encrypt-data-using-openssl-rsa.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        - bytes: 60 86 48 01 65 03 04 03 0D = RSA-SHA3-224
```
[the remark this answers is not in the document]
Comment (Ana06) [reply nursery/encrypt-data-using-openssl-rsa.yml]: @williballenthin have you tested that capa works with this sample? I am running current capa master and it doesn't like this sample
Screenshot 2022-12-20 at 20 01 01

**Review by mr-tz — APPROVED** [review]

**Pull request merged**